From 09c4aa92a8a1be78d3ac127ba94ae0d6c7a624cd Mon Sep 17 00:00:00 2001
From: Jon Turney <jon.turney@dronecode.org.uk>
Date: Thu, 8 Oct 2015 02:11:41 +0100
Subject: [PATCH] Fix building client library and upload tools for MinGW

This is the rest of https://breakpad.appspot.com/548002/, brought up to date

v2:
Refine MinGW changes in HTTPUpload::GetFileContents so it closes file after use

v3:
For consistency, write conditionals in terms of _MSC_VER, not __MINGW32__

v4:
Use fd rather than FILE * in HTTPUpload::GetFileContents

It appears that constructing a stdio_filebuf from a FILE * does a fflush(0),
which has been seen to occasionally fail EBADF (on at least W7 x64).

Both of these things seem like they might be bugs

Workaround for the moment by constructing stdio_filebuf from a fd instead.

v5:
Drop HTTPUpload::GetFileContents() changes as upstream now uses WideTOMBCP()
Drop changes to avoid stdext::checked_array_iterator, as upstream

Signed-off-by: Jon Turney <jon.turney@dronecode.org.uk>
---
 .../windows/tests/crash_generation_app/crash_generation_app.cc  | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/client/windows/tests/crash_generation_app/crash_generation_app.cc b/src/client/windows/tests/crash_generation_app/crash_generation_app.cc
index 7bc22286..1fa5f58d 100644
--- a/src/client/windows/tests/crash_generation_app/crash_generation_app.cc
+++ b/src/client/windows/tests/crash_generation_app/crash_generation_app.cc
@@ -483,9 +483,11 @@ int APIENTRY _tWinMain(HINSTANCE instance,
   CustomClientInfo custom_info = {kCustomInfoEntries, kCustomInfoCount};
 
   CrashServerStart();
+#ifdef _MSC_VER
   // This is needed for CRT to not show dialog for invalid param
   // failures and instead let the code handle it.
   _CrtSetReportMode(_CRT_ASSERT, 0);
+#endif
   handler = new ExceptionHandler(L"C:\\dumps\\",
                                  nullptr,
                                  google_breakpad::ShowDumpResults,
-- 
2.53.0.windows.1

